jobs/aadAppRegistration.yml

```diff
@@ -22,6 +24,25 @@ jobs:
   - checkout: self  # self represents the repo where the initial Pipelines YAML file was found
     clean: true  # whether to fetch clean each time
 
+  - task: GitVersion@4
+    displayName: GitVersion
+    inputs:
+      preferBundledVersion: false
+
+  - powershell: | 
+      if ('${{ parameters.allowedEnvironments }}' -match '$(environmentName)') {Write-Host "Variable validation successful!"}
+      else {Write-Error ("Variable validation failed! Please check if environment {0} is from allowed environmenst {1}" -f '$(environmentName)', '"${{ parameters.allowedEnvironments }}"')}
+      
+      if ('refs/heads/master' -match '$(Build.SourceBranch)' -and '${{ parameters.productionLikeEnvironments }}' -match '$(environmentName)'){
+      Write-Host ("Deployig from '{0}' to production like environment '{1}'..." -f '$(Build.SourceBranch)', '$(environmentName)')}
+
+      elseif("refs/heads/master" -notmatch '$(Build.SourceBranch)' -and '${{ parameters.productionLikeEnvironments }}' -match '$(environmentName)'){
+      Write-Error ("Deployig from '{0}' to non-production like environment '{1}'. Make sure you deploy from master branch to production like environments. " -f '$(Build.SourceBranch)', '$(environmentName)')}
+      
+      elseif("refs/heads/master" -notmatch '$(Build.SourceBranch)' -and '${{ parameters.productionLikeEnvironments }}' -notmatch '$(environmentName)'){
```

Comment:
Just a thought. I think most of the time any if+elseif needs an else. Otherwise you have like:
if x elseif y but what happens in case z, w etc that you haven't thought of? Oftentimes it's good to catch just for easier error handling.

Might not be the case here.